### ISSUE TITLE
Error in timezone configuration

### ISSUE THREAD

**Issue by cjuniorfox** [AzuraCast/AzuraCast#523]
- **Installation Method:** 
Docker
- **Host OS (if using Docker):** 
Ubuntu 16.04
**Describe the issue in full detail, including screenshots if possible:**
When I try to schedule a some playlist into some time period, my lists never play at the time as desired, aways delaying one hour.
I'm from Rio de Janeiro. UTC-3. I configure that way on "edit my profile", but the azuracast works as like I configured UTC-4 and not UTC-3.

Example. Now as I writing this bug, is 8:10 AM UTC -3. The GMT time is 11:10.
So, I edit my profile, change the time zone to GMT-3 (Atlantic Time) and on this page (edit profile), date and time seems right to me, but when I save and edit playlist, the scheduler shows scheduled like GMT-4. Show current date and time correctly, but scheduler not works on that way.
He shows 7:00 AST (11:00 AM UTC).
If setup schenduler referencing on UTC time, all works ok, but when I use time zone based configuration I have the issue.

I uploaded some screenshots
Regards

![image](https://user-images.githubusercontent.com/3306393/37903268-a22b4f46-30cd-11e8-9ce4-84ebb9ad291f.png)

![image](https://user-images.githubusercontent.com/3306393/37903324-d34e2e40-30cd-11e8-95fd-8d7ce1bde1ed.png)

**Comment by dreinn** [comment]
I confirm, for example on my web station, it says that server time is 08.27 AM, instead server time is 10.27AM.
I can change my profile UTC settings, but server time is not adjustable and so playlists start at wrogn time

**Comment by SlvrEagle23** [comment]
@cjuniorfox @dreinn As any developer will tell you, time can sometimes be a difficult thing.

In this case, after playing with it myself, I quickly discovered the mistake that's causing this issue: the "time code" system used by playlists is basing their time as if it was hours on January 1, when DST is inactive for a large majority of time zones. Thus, if you're in an area where DST is active, the "Your local time" message would display one time offset, and the dropdowns would display another, which is indeed very confusing.

I've reworked the time-displaying part of the code to reference the hour as if it's on the same day, month and year as the current date, which fixes a large majority of those issues.

**Comment by dreinn** [comment]
I updated with new code but unfortunately nothing is changed for me.
Azuracast is at 10.00, realtime is 12.00.
I don't know if this can help you, but when I update the repository and run the docker-compose command:

docker-compose run --rm cli azuracast_update

the reported time is already wrong, so station server start at 8.35 instead it is 10.35.

**Comment by dreinn** [comment]
yeah!
I think it works for me!
I went to my account profile and set to utc +2.00.
So now when I want to add a playlist, I see the exact time and the normal utc time.
And after creating a playlist that started at 16.00, ti works correctly!
Thank youuu!

**Comment by cjuniorfox** [comment]
I Think works for me too!
Now is scheduling correctly and appears to be playing correctly too.
Just one little issue has been left (I not reported earlier). It's about schedule view.
If some playlist is scheduled crossing midnight, the schedule view doesn't work. Looking into debugging window, fells like you're using some time and date plugin, and it's setting always 1/1/2018 as date.
If some list is scheduled between 11:30PM and 12:30AM, he tryes to use this plugin passing time and date starting 1/1/2018 - 11:30 PM and ending 1/1/2018 - 12:30 AM. This is impossible and throws a exception hanging the functionality.

I'll open another issue and close this one. Thank you for your fast fixing.

**Comment by cjuniorfox** [comment]
I Closed the issue, but reopened because I realize the playlist is starting correctly, but ending wrong.
Example, I have one playlist starting at 20:00 PM (AST) 11:00 PM UTC and ending at 22:00 PM (AST) 1:00 AM UTC.
But the playlist stopped playing one hour earlier, at 21:00 PM (AST) 00:00 UTC.

**Comment by SlvrEagle23** [comment]
@cjuniorfox Did you set the playlist to play only on certain days? If you set a playlist to play on certain days, it'll play when the playlist falls within those time periods on that day as specified by UTC. There's no way for us to make it play based on the days in your user time zone, as the server has to have a single time zone it always uses for calculations, so it uses UTC as a standard.

**Comment by SlvrEagle23** [comment]
@cjuniorfox I've updated the playlist processing to make a lot more intuitive sense, so when you set a playlist to play on a given day of the week, and it goes overnight, it starts on the day you specified and continues into the _next_ morning, instead of splitting into "midnight to 4am on that day" and "8pm to midnight on that same day", which makes no sense at all.

**Comment by SlvrEagle23** [comment]
@cjuniorfox Any updates? Let me know if you still need help with this issue.

**Comment by cjuniorfox** [comment]
@SlvrEagle23 , first of all, thank you for your effort to increase the quality of Azuracast.
The platform is superb.
I updated Azuracast today and had a little issue to update, where I had to start manually Icecast and LiquidSoap services, but despite that, the radio service still working ok.

Because I setup to play this playlist over the week, I'll give feedback about this issue Tuesday.
Anyway, thanks again for your effort.

Regards.
